File: 11_multithreading/python/naive_count.py
import threading

def thread_plus(name):
    global data
    for _ in range(100_000):
        data += 1

if __name__ == "__main__":
    data = 0
    threads = list()
    for index in range(10):
        x = threading.Thread(target=thread_plus, args=(index,))
        threads.append(x)
        x.start()

    for h in threads:
        h.join()
    print(data)

File: 11_multithreading/rust/naive_count.rs
fn main() {
    let mut data: u32 = 0;
    let mut handles = vec![];

    for _ in 0..10 {
        let h = std::thread::spawn(move || {
            for _ in 0..100_000 {
                data += 1;
            }
            println!("{}", data);
        });

        handles.push(h);
    }
    for h in handles {
        h.join().unwrap();
    }
    println!("{}", data);
}
